File: aula_neville.py
def neville(count, pontos, x, Neville):
	j = 1

	#Calculo da matriz de Neville
	while j < count:
		i = 0
		while i < (count-j):
			Neville[j].append((((x-pontos[0][j+i])*Neville[j-1][i])-((x-pontos[0][i])*Neville[j-1][i+1]))/(pontos[0][i]-pontos[0][i+j]))
			i += 1
		j+=1

	return Neville

File: aula_neville.c
float Neville(int n, float pontos[n][2], float x, float neville[n][n])
{
  int i, j;

    for(j=1; j<n; j++)
    {
      for(i=0;i<(n-j);i++)
      {
      	//Equação característica da interpolação pelo método de Neville
	     neville[i][j]=(((x-pontos[j+i][0])*neville[i][j-1])-((x-pontos[i][0])*neville[i+1][j-1]))/(pontos[i][0]-pontos[i+j][0]);
      }
    }
}
